### PR TITLE
Chore/edc 0.7.0 iatp mock

### PR DESCRIPTION
**Pull request by ds-jhartmann**
## Description

- Add iatp mock and helm chart
- update tx-data-provider EDC to 0.7.1
- switch to fixed key/cert for EDC which can be used for iatp mock
- add bdrs memory service + seeding with BPN + DIDs

https://github.com/eclipse-tractusx/sig-release/issues/710

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: charts/umbrella/values.yaml [hunk 1]
```diff
@@ -691,16 +596,25 @@ dataconsumerOne:
...
+    iatp:
+      id: did:web:mock-util-service/BPNL00000003AZQP
+      trustedIssuers:
+        - did:web:mock-util-service/trusted-issuer
+      sts:
+        dim:
+          url: http://mock-util-service/sts
         oauth:
-          tokenurl: http://centralidp.tx.test/auth/realms/CX-Central/protocol/openid-connect/token
+          token_url: http://umbrella-centralidp:80/auth/realms/CX-Central/protocol/openid-connect/token
```
Comment (evegufy) [inline charts/umbrella/values.yaml:607]: why the change to the internal service?

File: iatp-mock/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+| Service           | address                                           | reachable where          |
+|-------------------|---------------------------------------------------|--------------------------|
+| mock-util-service | http://mock-util-service:8080/edr-log             | docker compose (miw net) |
+| mock-util-service | http://mock-util-service:8080/sts                 | docker compose (miw net) |
+| mock-util-service | http://mock-util-service:8080/presentations/query | docker compose (miw net) |
+| mock-util-service | http://mock-util-service:8080/{did-name}          | docker compose (miw net) |
+| mock-util-service | localhost:8888/*                                  | host machine             |
+
+Use the service to either simulate the central IAM or just to got some logging api
```
Comment (evegufy) [inline iatp-mock/README.md:44]: the mock isn't able to replace the centralidp keycloak, right?
just to avoid confusion it might be better to name the mock dim (decentral identity management) mock not iam mock.

File: charts/umbrella/README.md [hunk 3]
```diff
@@ -664,6 +664,12 @@ Build fuseki docker image by following the below steps:
 - Unzip the jena-fuseki-docker-4.7.0.zip.
 - Build the docker image by running the command - `docker build --build-arg JENA_VERSION=4.7.0 -t jena-fuseki-docker:4.7.0 --platform linux/amd64 .`
 
+## Precondition for IATP Mock
+
+in case of enabling `iatpmock` (e.g. by using [values-adopter-data-exchange.yaml](values-adopter-data-exchange.yaml)), the iatp-mock docker image must be built first:
+
+`docker build iatp-mock/ -t tractusx/iatp-mock:testing --platform linux/amd64`
```
Comment (evegufy) [inline charts/umbrella/README.md:671]: we should discuss if it's not worth to also publish the image, for e2e testing maybe not necessary but from a local sandbox (easy entrance point for new developers) maybe beneficial

**Review by evegufy — CHANGES_REQUESTED** [review]
Looks really good! Just some comments and when installing locally with dataexchange subset, I encounter the following issue:
Error: INSTALLATION FAILED: execution error at (umbrella/charts/tx-data-provider/charts/tractusx-connector/templates/deployment-controlplane.yaml:302:75): .Values.backendService.httpProxyTokenReceiverUrl is required

File: charts/umbrella/README.md (hunk 3 above)
Comment (ds-jhartmann) [reply charts/umbrella/README.md:671]: That's something I don't like either. The mock requires configuration in code which can change depending on the use-case. I did not manage to get this configurable via helm chart in a reasonable amount of time, so I decided to not publish the image. 
I'm open for suggestions here

File: iatp-mock/README.md (hunk 2 above)
Comment (ds-jhartmann) [reply iatp-mock/README.md:44]: Right, I missed this in the Readme. I went with the name "IATP mock", since this is what the mock is doing and it should be distinguishable enough from IAM

File: charts/umbrella/values.yaml (hunk 1 above)
Comment (ds-jhartmann) [reply charts/umbrella/values.yaml:607]: For some reason the ingress doesn't work inside the cluster of the workflow. Not sure, if this worked previously, but to get the EDC to work, I had to go back to the internal service names. 
As discussed, I changed it back to the ingress URL and use the service URL only for the testing config

File: charts/umbrella/Chart.yaml [hunk 4]
```diff
@@ -75,22 +70,32 @@ dependencies:
...
+    version: 0.1.0
+    repository: file://../tx-data-provider
     condition: dataconsumerTwo.enabled
     # pgadmin4 as helper tool for easy database access
   - condition: pgadmin4.enabled
     name: pgadmin4
     repository: https://helm.runix.net
     version: 1.25.x
+    # bdrs
+  - name: bdrs-server-memory
```
[the remark this answers is not in the document]
Comment (ds-jhartmann) [reply charts/umbrella/Chart.yaml:93]: For easier configuration, I went with the in-memory version and for now I would prefer to stick with the in-memory variant for the time being with the possibility to change to the persistent variant in the future. Especially since the BDRS data is seeded at start-up, so no initial configuration is required.

File: charts/umbrella/Chart.yaml [hunk 5]
```diff
@@ -75,22 +70,32 @@ dependencies:
     # TX Data Consumer 1
   - name: tx-data-provider
     alias: dataconsumerOne
-    version: 0.0.5
-    repository: https://eclipse-tractusx.github.io/charts/dev
+    version: 0.1.0
+    repository: file://../tx-data-provider
```
[the remark this answers is not in the document]
Comment (ds-jhartmann) [reply charts/umbrella/Chart.yaml:74]: Right. Since we plan on getting the 24.05 release to a working state first, I went with this approach to be able to test the full data exchange flow

File: charts/umbrella/README.md (hunk 3 above)
Comment (evegufy) [reply charts/umbrella/README.md:676]: @tom-rm-meyer-ISST could you please have a look if it's possible to make the mock more configurable?

**Comment by evegufy** [comment]
> > Looks really good! Just some comments and when installing locally with dataexchange subset, I encounter the following issue: Error: INSTALLATION FAILED: execution error at (umbrella/charts/tx-data-provider/charts/tractusx-connector/templates/deployment-controlplane.yaml:302:75): .Values.backendService.httpProxyTokenReceiverUrl is required
> 
> Regarding the local installation: I could not reproduce this on my machine. Did you update the depencencies of the tx-data-provider chart? Since the EDC version was updated, the charts have to be pulled first. I could install it with
> 
> ```
> helm dep up .\charts\tx-data-provider\
> helm dep up .\charts\umbrella\
> helm install umbrella .\charts\umbrella\ -n umbrella --create-namespace -f .\charts\umbrella\values-adopter-data-exchange.yaml
> ```

HI @ds-jhartmann thanks I was able to install it but the dataprovider testdata postinstall is failing, that's not the case for you?
[logs-from-post-install-job-in-umbrella-dataprovider-post-install-testdata-rl89g.log](https://github.com/user-attachments/files/16232275/logs-from-post-install-job-in-umbrella-dataprovider-post-install-testdata-rl89g.log)

**Comment by evegufy** [comment]
just commenting that I noticed the following warning as install

```
coalesce.go:289: warning: destination for tractusx-connector.vault.injector.webhook.objectSelector is a table. Ignoring non-table value (matchExpressions:
- key: app.kubernetes.io/name
  operator: NotIn
  values:
  - {{ template "vault.name" . }}-agent-injector
)
coalesce.go:289: warning: destination for tractusx-connector.vault.injector.webhook.objectSelector is a table. Ignoring non-table value (matchExpressions:
- key: app.kubernetes.io/name
  operator: NotIn
  values:
  - {{ template "vault.name" . }}-agent-injector
)
```
as discussed, they appear to be inherent to the new edc version in this install scenario

**Review by evegufy — APPROVED** [review]
as discussed in the last open meeting, it would be great to prepare for easy image override, I hope it's fine that I pushed the change https://github.com/eclipse-tractusx/tractus-x-umbrella/pull/101/commits/cc5200e959ed8d0a12acbfb27d057014f90234ad

**Comment by tom-rm-meyer-ISST** [comment]
For some reason my comment was still 'pending'. 

@ds-jhartmann @evegufy I had a check on that as I've not yet done a configuration for python. I think I would be able to achieve a configuration in either of the two ways in around < 0.5PD. The options below only focus on the docker environment vars.

Sidenotes: 
- It may make sense to configure only one trusted issuer, as this should be sufficient. Further the ES256 keys need to be generated just once and are used for all partners.
- One need to mount the key directory with their names accordingly and need to put the private + public keys into the vaults as the EDC accesses them.

As some time went by, do we need further configuration options? How should I contribute that? First in PURIS or directly with a branch forked from this branch so that you can merge? I maybe can manage to get this within this or latest by end of next week. I can also adapt it in the puris repo so that one of you can review it and see the delta easily.

# Option A: configure participants and their key via json array in one environment var 
```
PARTICIPANTS='[
       {
           "did": "did:web:mock-util-service/trusted-issuer",
           "bpnl": "NONE",
           "did_resolve_name": "trusted-issuer",
           "kid_vault": "",
           "private_key_path": "",
           "is_trusted_issuer": true
       },
       {
           "did": "did:web:mock-util-service/supplier",
           "bpnl": "BPNL1234567890ZZ",
           "did_resolve_name": "supplier",
           "kid_vault": "supplier-cert",
           "private_key_path": "keys/supplier.key",
           "is_trusted_issuer": false
       },
       {
           "did": "did:web:mock-util-service/customer",
           "bpnl": "BPNL4444444444XX",
           "did_resolve_name": "customer",
           "kid_vault": "customer-cert",
           "private_key_path": "keys/customer.key",
           "is_trusted_issuer": false
       }
   ]'
```

# Option B: configure participants dynamically using prefix
```
PARTICIPANT_1_DID="did:web:mock-util-service/trusted-issuer"
PARTICIPANT_1_BPNL="NONE"
PARTICIPANT_1_DID_RESOLVE_NAME="trusted-issuer"
PARTICIPANT_1_KID_VAULT=""
PARTICIPANT_1_PRIVATE_KEY_PATH=""
PARTICIPANT_1_IS_TRUSTED_ISSUER=true

PARTICIPANT_2_DID="did:web:mock-util-service/supplier"
PARTICIPANT_2_BPNL="BPNL1234567890ZZ"
PARTICIPANT_2_DID_RESOLVE_NAME="supplier"
PARTICIPANT_2_KID_VAULT="supplier-cert"
PARTICIPANT_2_PRIVATE_KEY_PATH="keys/supplier.key"
PARTICIPANT_2_IS_TRUSTED_ISSUER=false

PARTICIPANT_3_DID="did:web:mock-util-service/customer"
PARTICIPANT_3_BPNL="BPNL4444444444XX"
PARTICIPANT_3_DID_RESOLVE_NAME="customer"
PARTICIPANT_3_KID_VAULT="customer-cert"
PARTICIPANT_3_PRIVATE_KEY_PATH="keys/customer.key"
PARTICIPANT_3_IS_TRUSTED_ISSUER=false
```

**Comment by ds-jhartmann** [comment]
Hi @tom-rm-meyer-ISST thank you for the input. 
I like Option A a bit better, I think it makes it a bit easier to add additional participants.
It would be great, if you could contribute this directly to the umbrella repo.

**Comment by tom-rm-meyer-ISST** [comment]
> Hi @tom-rm-meyer-ISST thank you for the input. I like Option A a bit better, I think it makes it a bit easier to add additional participants. It would be great, if you could contribute this directly to the umbrella repo.

Will try to give it a shot after QGate / R24.08. Which branch should I use to fork from? this one or upgrade/24.05?
